sam-build-and-deploy/action.yml

```diff
@@ -0,0 +1,58 @@
+name: SAM Build and Deploy
+description: SAM Build and Deploy
+
+inputs:
+  environment:
+    description: Environment
+    required: true
+  microservice-name:
+    description: Microservice name
+    required: true
+  aws-account-id:
+    description: AWS account id
+    required: true
+  aws-access-key-id:
+    description: AWS access key id
+    required: true
+  aws-secret-access-key:
+    description: AWS secret access key
+    required: true
+  aws-region:
+    description: AWS Region
+    required: true
+  s3-bucket:
+    description: S3 bucket
+    required: false
+    default: doo-infrastructure
+
+runs:
+  using: composite
+  steps:
+    - uses: aws-actions/setup-sam@v1
+    - uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ inputs.aws-access-key-id }}
+        aws-secret-access-key: ${{ inputs.aws-secret-access-key }}
+        aws-region: ${{ inputs.aws-region }}
+
+    - id: build-image
+      name: Build image
+      shell: bash
+      run: >
+        sam build
+        --base-dir .
+        --template-file ./infrastructure/infrastructure.yml
+    - id: deploy-stack
+      name: Deploy stack

```

Comment:
Please put the name as the first item of the step configuration on each step, then ID (if needed) and the rest:

```
- name: Deploy stack
   id: deploy-stack
   ...
```

Also please add one empty line between each step in the file.